core/src/main/java/com/yahoo/ycsb/BasicDB.java

```diff
@@ -21,280 +21,249 @@
-		{
-			System.out.println("***************** properties *****************");
-			Properties p=getProperties();
-			if (p!=null)
-			{
-				for (Enumeration e=p.propertyNames(); e.hasMoreElements(); )
-				{
-					String k=(String)e.nextElement();
-					System.out.println("\""+k+"\"=\""+p.getProperty(k)+"\"");
-				}
-			}
-			System.out.println("**********************************************");
-		}
-	}
-
-  static final ThreadLocal<StringBuilder>  TL_STRING_BUILDER = new ThreadLocal<StringBuilder>() {
+public class BasicDB extends DB {
+  public static final String VERBOSE = "basicdb.verbose";
+  public static final String VERBOSE_DEFAULT = "true";
+
+  public static final String SIMULATE_DELAY = "basicdb.simulatedelay";
+  public static final String SIMULATE_DELAY_DEFAULT = "0";
+
+  public static final String RANDOMIZE_DELAY = "basicdb.randomizedelay";
+  public static final String RANDOMIZE_DELAY_DEFAULT = "true";
+
+  private boolean verbose;
+  private boolean randomizedelay;
+  private int todelay;
+
+  public BasicDB() {
+    todelay = 0;
+  }
+
+  private void delay() {
+    if (todelay > 0) {
+      long delayNs;
+      if (randomizedelay) {
+        delayNs = TimeUnit.MILLISECONDS.toNanos(Utils.random().nextInt(todelay));
+        if (delayNs == 0) {
+          return;
+        }
+      } else {
+        delayNs = TimeUnit.MILLISECONDS.toNanos(todelay);
+      }
+
+      final long deadline = System.nanoTime() + delayNs;
+      do {
+        LockSupport.parkNanos(deadline - System.nanoTime());
+      } while (System.nanoTime() < deadline && !Thread.interrupted());
+    }
+  }
+
+  /**
+   * Initialize any state for this DB.
+   * Called once per DB instance; there is one DB instance per client thread.
+   */
+  @SuppressWarnings("unchecked")
+  public void init() {
+    verbose = Boolean.parseBoolean(getProperties().getProperty(VERBOSE, VERBOSE_DEFAULT));
+    todelay = Integer.parseInt(getProperties().getProperty(SIMULATE_DELAY, SIMULATE_DELAY_DEFAULT));
+    randomizedelay = Boolean.parseBoolean(getProperties().getProperty(RANDOMIZE_DELAY, RANDOMIZE_DELAY_DEFAULT));
+    if (verbose) {
+      synchronized (System.out) {
+        System.out.println("***************** properties *****************");
+        Properties p = getProperties();
+        if (p != null) {
+          for (Enumeration e = p.propertyNames(); e.hasMoreElements();) {
+            String k = (String) e.nextElement();
+            System.out.println("\"" + k + "\"=\"" + p.getProperty(k) + "\"");
+          }
+        }
+        System.out.println("**********************************************");
+      }
+    }
+  }
+
+  private static final ThreadLocal<StringBuilder> TL_STRING_BUILDER = new ThreadLocal<StringBuilder>() {
     @Override
     protected StringBuilder initialValue() {
       return new StringBuilder();
     }
   };
 
-  static StringBuilder getStringBuilder() {
+  private static StringBuilder getStringBuilder() {
```

Comment:
Super nit-picky but if anyone had their own code that was calling this from within the package or their own sublass then it would be a breaking change. Probably not a big deal at all.